mlrun/datastore/targets.py

```diff
@@ -371,11 +399,53 @@ def get_spark_options(self, key_column=None, timestamp_key=None):
 
 class ParquetTarget(BaseStoreTarget):
     kind = TargetTypes.parquet
-    suffix = ".parquet"
     is_offline = True
     support_spark = True
     support_storey = True
 
+    def __init__(
+        self,
+        name: str = "",
+        path=None,
+        attributes: typing.Dict[str, str] = None,
+        after_state=None,
+        columns=None,
+        partitioned: bool = False,
+        key_bucketing_number: typing.Optional[int] = None,
+        partition_cols: typing.Optional[typing.List[str]] = None,
+        time_partitioning_granularity: typing.Optional[str] = None,
+    ):
+        super().__init__(
+            name,
+            path,
+            attributes,
+            after_state,
+            columns,
+            partitioned,
+            key_bucketing_number,
+            partition_cols,
+            time_partitioning_granularity,
+        )
+
+        if (
+            time_partitioning_granularity is not None
+            and time_partitioning_granularity not in self._legal_time_units
+        ):
+            raise ValueError(
+                f"time_partitioning_granularity parameter must be one of {','.join(self._legal_time_units)}, "
+                f"not {time_partitioning_granularity}."
+            )
+
+        self.suffix = (
+            ".parquet"
+            if not partitioned
+            and [key_bucketing_number, partition_cols, time_partitioning_granularity]
+            == [None] * 3
+            else ""
+        )
+
+    _legal_time_units = ["year", "month", "day", "hour", "minute", "second"]
```

Comment:
Should be imported from storey, no?

Reply:
storey doesn't have this list. I'm not sure putting it in storey just for the sake of importing it is so great. I mean, it makes some logical sense, but creates another sync point between the projects unnecessarily.

Reply:
Added the list in https://github.com/mlrun/storey/pull/219
But mlrun tests fail on import from storey